packages/relay/tests/lib/formatters.spec.ts

```diff
@@ -325,6 +330,16 @@ describe('Formatters', () => {
       const formattedResult: any = formatContractResult({ ...contractResult, chain_id: '0x' });
       expect(formattedResult.chainId).to.be.undefined;
     });
+
+    it('Should return legacy EIP155 transaction when null type', () => {
+      const formattedResult: any = formatContractResult({ ...contractResult, type: null });
+      expect(formattedResult.type).to.be.eq('0x0');
+    });
+
+    it('Should return null when contract result type is undefined', async function () {
+      const formattedResult = formatContractResult({ ...contractResult, type: undefined });
+      expect(formattedResult).to.be.null;
+    });
```

Comment:
QQ: Is this the behavior that we want? Why do we treat `null` and `undefined` differently here?

Reply:
That’s a good question. Unfortunately, this method is used in a couple of places, and I didn’t want to change it without further research, which is out of the scope of this PR.